### DISCUSSION TITLE
Gain focus from killing enemies

### DISCUSSION THREAD

**Discussion by Krwak** [cataclysmbn/Cataclysm-BN#2800] [General]
### Is your feature request related to a problem? Please describe.

I have been thinking about how to make combat more interesting and fast paced and one thing that i came up with is that focus goes down really rapidly during combat which rewards frequent stops to refill focus in order to increase skill gains.
One solution for this would be to reward players for continuing the fight in some way, instead of punishing them for it, despite all the wounds.
A typical example is to hunker in a vehicle to listen music and read a book or play some game for a few hours until focus replenishes, then back to combat again.

### Describe the solution you'd like

Not sure how feasible it would be:
If survivor kills certain amount of zombies during let's say 5 -10 minutes, they will receive a flat focus increase.
Not morale bonus as that one is pretty marginal.
Not sure about numbers yet but curious whether this would be possible or not.

**Comment by scarf005** [comment]
that'd be a nice addition. additionally, how about giving a small morale for killing tough enemies?

**Comment by metalheimer** [comment]
Gave me the idea that baseline focus should drop slowly over the day. Sleep 8h. Next 8h good baseline focus, never dropping below 70. Next 8h, allow dropping below 70, possibly disallow above 100 unless with heavy stimulants.

Something like that.

**Comment by Krwak** [comment]
> that'd be a nice addition. additionally, how about giving a small morale for killing tough enemies?

That might work, not sure if it makes any kind of difference though. Combat seems to be so focus intensive that it drains really fast no matter what. Would be good to have it nonetheless.

**Comment by scarf005** [comment]
i wonder which part makes combat focus draining, should also lessen them

**Comment by Ignaramico** [comment]
what about the trait "killing drive" (or something, can recall the name), i havent use it, but i feel that it should replenish/boost focus and not just give moral

**Comment by Krwak** [comment]
> what about the trait "killing drive" (or something, can recall the name), i havent use it, but i feel that it should replenish/boost focus and not just give moral

That's one possibility how to test this mechanic, in case somebody decides to implement it.
Currently killer drive is really not worth the 2 points and the small morale boost you get from killing enemies is negligible.
There's even a morale debuff that increases if you don't kill an enemy for 24 hours and longer.

**Comment by metalheimer** [comment]
> i wonder which part makes combat focus draining, should also lessen them

Probably because it involves multiple skills. In melee: dodging, melee skill, and often two weapon skills because melee weapons often do multiple types of damage.

**Comment by chaosvolt** [comment]
This leads me to a question: do the functions that practice skills in the code, like during combat, have any sort of boolean that could be toggled to make it not affect (or be affected by, ideally) focus? This would fix it without any big code changes, otherwise we'd have to build support for granting skill EXP without focus being taken into account.

If we ever implement the athletics skill idea we're going to likely need that anyway, since automatically training a skill any time you take any action that burns stamina is going to mandate not having it mess with your focus (plus the EXP here would represent progress in physical conditioning using a mechanic more natural to the game than DDA's cardio system, so it being treated like the PC learning how to "run more gooder" wouldn't make sense).
